store/sqlstore/channel_store.go

```diff
@@ -3316,45 +3363,111 @@ func (s SqlChannelStore) buildFulltextClause(term string, searchColumns string)
 	return
 }
 
-func (s SqlChannelStore) performSearch(searchQuery string, term string, parameters map[string]interface{}) (model.ChannelList, error) {
-	likeClause, likeTerm := s.buildLIKEClause(term, "c.Name, c.DisplayName, c.Purpose")
-	if likeTerm == "" {
-		// If the likeTerm is empty after preparing, then don't bother searching.
-		searchQuery = strings.Replace(searchQuery, "SEARCH_CLAUSE", "", 1)
-	} else {
-		parameters["LikeTerm"] = likeTerm
-		fulltextClause, fulltextTerm := s.buildFulltextClause(term, "c.Name, c.DisplayName, c.Purpose")
-		parameters["FulltextTerm"] = fulltextTerm
-		searchQuery = strings.Replace(searchQuery, "SEARCH_CLAUSE", "AND ("+likeClause+" OR "+fulltextClause+")", 1)
+func (s SqlChannelStore) squirrelBuildFulltextClause(term string, searchColumns string) (fulltextClause sq.Sqlizer) {
+	// Copy the terms as we will need to prepare them differently for each search type.
+	fulltextTerm := term
+
+	// These chars must be treated as spaces in the fulltext query.
+	for _, c := range spaceFulltextSearchChar {
+		fulltextTerm = strings.Replace(fulltextTerm, c, " ", -1)
+	}
+
+	// Prepare the FULLTEXT portion of the query.
+	if s.DriverName() == model.DatabaseDriverPostgres {
+		fulltextTerm = strings.Replace(fulltextTerm, "|", "", -1)
+
+		splitTerm := strings.Fields(fulltextTerm)
+		for i, t := range strings.Fields(fulltextTerm) {
+			if i == len(splitTerm)-1 {
+				splitTerm[i] = t + ":*"
+			} else {
+				splitTerm[i] = t + ":* &"
+			}
+		}
+
+		fulltextTerm = strings.Join(splitTerm, " ")
+
+		cols := convertMySQLFullTextColumnsToPostgres(searchColumns)
+		fullClauseFmt := "((to_tsvector('english', %s)) @@ to_tsquery('english', ?))"
+
+		fulltextClause = sq.Expr(fmt.Sprintf(fullClauseFmt, cols), fulltextTerm)
+	} else if s.DriverName() == model.DatabaseDriverMysql {
+		splitTerm := strings.Fields(fulltextTerm)
+		for i, t := range strings.Fields(fulltextTerm) {
+			splitTerm[i] = "+" + t + "*"
+		}
+
+		fulltextTerm = strings.Join(splitTerm, " ")
+
+		fullClauseFmt := "MATCH(%s) AGAINST (? IN BOOLEAN MODE)"
+		fulltextClause = sq.Expr(fmt.Sprintf(fullClauseFmt, searchColumns), fulltextTerm)
+	}
+
+	return
+}
+
+func (s SqlChannelStore) performSearch(searchQuery sq.SelectBuilder, term string) (model.ChannelList, error) {
+	var sql string
+	var parameters []interface{}
+	sql, parameters, err := searchQuery.ToSql()
+	_ = parameters
+	_ = sql
+	if err != nil {
+		return model.ChannelList{}, errors.Wrapf(err, "failed to construct query with term='%s'", term)
 	}
 
 	var channels model.ChannelList
 
-	if _, err := s.GetReplica().Select(&channels, searchQuery, parameters); err != nil {
-		return nil, errors.Wrapf(err, "failed to find Channels with term='%s'", term)
+	err2 := s.GetReplicaX().Select(&channels, sql, parameters...)
+	if err2 != nil {
```

Comment:
I believe this can be moved back into the previous statement like it was originally.